rdf/shared/src/main/scala/org/w3/banana/io/RDFQuadReader.scala

```diff
@@ -0,0 +1,21 @@
+package org.w3.banana.io
+
+import java.io.{InputStream, Reader}
+
+import org.w3.banana.RDF
+
+trait RDFQuadReader[Rdf <: RDF, M[_], +S] extends RDFReader[Rdf, M, S] {
+
+  def readAll(is: InputStream, base: String): M[Map[Option[Rdf#Node], Rdf#Graph]]
+
+  def readAll(in: Reader, base: String): M[Map[Option[Rdf#Node], Rdf#Graph]]
+
+  def read(is: InputStream, base: String, graphName: Rdf#URI): M[Rdf#Graph]
+
+  def read(in: Reader, base: String, graphName: Rdf#URI): M[Rdf#Graph]
+
+  def readDefaultGraph(is: InputStream, base: String): M[Rdf#Graph]
+
+  def readDefaultGraph(in: Reader, base: String): M[Rdf#Graph]
```

Comment:
When I first implemented the interface, I was thinking  of bigger graphs, where filtering during parse time could make sense. Unfortunately I couldn't find a solution in Jena and Sesame to do so. So I agree, these methods aren't necessary, except they could reduce some temporary memory overhead.

Reply:
yes, I don't think those frameworks are very good for parsing huge graphs for that reason. The right way I think in the end to do this would be to have asynchronous parsers-  that need to be written -- to produce (akka) streams of sets of quads. Then one could easily add a filter to the Stream as they are designed for that. 

But in the meantime adding support for existing parsers provided by Jena and Sesame would be useful. But since they don't do it right, and to do it right one needs a completely different framework, I'd just remove those methods. Perhaps if they start offering filtering at some point then those methods could be added, but at present, this does not add anything and it makes one think one is getting some value where one is not.

Reply:
well actually if you see that they do use streams, then they are not that bad. They are just not as good as if they were non-blocking.